sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/SparkShreddingUtils.scala

```diff
@@ -188,6 +570,32 @@ case object SparkShreddingUtils {
       scalarSchema, objectSchema, arraySchema)
   }
 
+  // Convert a scalar variant schema into a Spark scalar type.
+  def scalarSchemaToSparkType(scalar: VariantSchema.ScalarType): DataType = scalar match {
+    case _: VariantSchema.StringType => StringType
```

Comment:
I wonder if we can have a util that lists supported types here .. otherwise, it's very likely we miss to fix this place when we happen to support more types.

Reply:
I think we can just keep it as it is because:
1. Essentially, this function only needs to process the `VariantSchema` returned by `buildVariantSchema`, which we have control of and will know it has changed.
2. I cannot think of an easy way list all supported types. Decimal types can have many precision-scale combinations.